onnxruntime/core/optimizer/attention_fusion.cc

```diff
@@ -571,4 +458,307 @@ bool AttentionFusion::FuseSubGraph(Node& layer_norm, const Node& add_after_layer
   return true;
 }
 
+static bool FuseSubGraphQKDistilBertImpl(Node& layer_norm, Graph& graph, AttentionFusionHelper::AttentionMaskNodesDistilBert& mask_nodes, const std::vector<std::reference_wrapper<const Node>>& parent_path_nodes,
+                                         int64_t hidden_size, int64_t num_heads, int64_t head_size, std::map<std::string, NodeArg*>& mask_index_map, const logging::Logger& logger) {
+  // path to q
+  std::vector<graph_utils::EdgeEndToMatch> q_path{
+      {0, 2, "MatMul", {1, 9, 13}, kOnnxDomain},
+      {0, 0, "Div", {7, 13}, kOnnxDomain},
```

Comment:
I think only difference is the Div node is before MatMul. There is no need to duplicate codes for distil bert. Just need add a check that Div can be before or after the MatMul node.